File: osrm-frontend/src/api/permission.ts
import request from './request'

export interface Permission {
  id: number
  parentId: number | null
  permissionCode: string
  permissionName: string
  resourceType: string
  action: string
  path: string
  icon: string
  sortOrder: number
  children?: Permission[]
}

export interface CreatePermissionData {
  parentId?: number
  permissionName: string
  permissionCode: string
  resourceType: string
  action?: string
  path?: string
  icon?: string
  sortOrder?: number
  description?: string
}

export const permissionApi = {
  // 获取权限树
  getTree() {
    return request.get('/v1/permissions/tree')
  },

  // 获取权限列表
  getList(params?: { permissionName?: string; permissionCode?: string; resourceType?: string }) {
    return request.get('/v1/permissions', { params })
  },

  // 创建权限
  create(data: CreatePermissionData) {
    return request.post('/v1/permissions', data)
  },

  // 删除权限
  delete(id: number) {
    return request.delete(`/v1/permissions/${id}`)
  }
}

File: osrm-backend/src/main/java/com/osrm/domain/user/entity/Permission.java
package com.osrm.domain.user.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "t_permission")
public class Permission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "parent_id")
    private Long parentId;

    @Column(name = "permission_code", nullable = false, unique = true, length = 100)
    private String permissionCode;

    @Column(name = "permission_name", nullable = false, length = 100)
    private String permissionName;

    @Column(name = "resource_type", nullable = false, length = 20)
    private String resourceType;

    @Column(name = "action", length = 20)
    private String action;

    @Column(name = "path", length = 500)
    private String path;

    @Column(name = "icon", length = 50)
    private String icon;

    @Column(name = "sort_order")
    private Integer sortOrder = 0;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getParentId() { return parentId; }
    public void setParentId(Long parentId) { this.parentId = parentId; }

    public String getPermissionCode() { return permissionCode; }
    public void setPermissionCode(String permissionCode) { this.permissionCode = permissionCode; }

    public String getPermissionName() { return permissionName; }
    public void setPermissionName(String permissionName) { this.permissionName = permissionName; }

    public String getResourceType() { return resourceType; }
    public void setResourceType(String resourceType) { this.resourceType = resourceType; }

    public String getAction() { return action; }
    public void setAction(String action) { this.action = action; }

    public String getPath() { return path; }
    public void setPath(String path) { this.path = path; }

    public String getIcon() { return icon; }
    public void setIcon(String icon) { this.icon = icon; }

    public Integer getSortOrder() { return sortOrder; }
    public void setSortOrder(Integer sortOrder) { this.sortOrder = sortOrder; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
}
